tests/unit/vdom.ts

```diff
@@ -2708,4 +2711,39 @@ describe('vdom', () => {
 			});
 		});
 	});
+
+	describe('behavior', () => {
+		it('will append nodes with attributes already attached', function(this: Test) {
+			if (!global.window || !global.window.hasOwnProperty('MutationObserver')) {
+				this.skip('MutationObserver not present');
+			}
+
+			class Foo extends WidgetBase {
+				render() {
+					return [v('div', { attr: 'test' }), v('div', { attr: 'test' })];
+				}
+			}
+
+			const doc = global.window.document;
+			const log: MutationRecord[] = [];
+			const observer = new MutationObserver((mutations: MutationRecord[]) => {
+				log.push(...mutations);
+			});
+
+			const parent = doc.createElement('div');
+			observer.observe(parent, {
+				attributes: true,
+				childList: true,
+				subtree: true
+			});
+			const widget = new Foo();
+			dom.append(parent, widget.__render__() as VNode, widget);
+
+			const results = [...log, ...observer.takeRecords()];
```

Comment:
Any reason you can't just do `log.push(...observer.takeRecords());` and use `log`?